aisim/prop.py

```diff
@@ -105,7 +105,7 @@ def transition(atoms, intensity_profile, tau, wave_vectors=None, wf=None, phase_
 
     propagator = np.array([[U_ee, U_eg], [U_ge, U_gg]], dtype='complex')
     propagator = np.transpose(propagator, (2, 0, 1))
-    # U*Psi
-    atoms.state_vectors = np.einsum('ijk,ki ->ij', propagator, atoms.state_vectors.T)
+    # U*|Psi>
+    atoms.state_vectors = np.conjugate(np.einsum('ijk,ki ->ij', propagator, np.conjugate(atoms.state_vectors).T))

```

Comment:
we should probably change the state vectors to be kets in the first place at some point or at least explicitly document the state vectors are bras.

Reply:
yes. this is a very good point I also had in mind.
Right now our state vectors are of shape (2,), so a 1D array.
Therefore transposing it has actually no effect.
Luckily for us, it seems working fine. I think thats because numpy.matmul and other array functions will treat a 1D array from the left side as a row vektor (bra) and on the right side as a column vector (ket). Therefore i did yet not find any problems due to the fact that we have a the state vector as a 1D array.
However, to be sure we make everything correct i would say we should name state_vectors-> state_bras and make them np.array([[a,b]]) (I think that has shape (1,2) and is a row vector). Than we can make the state_kets property which will just return
np.conjugate(state_bras).T and be of shape (2,1).
Then it will be rigorous.

Reply:
and we can use the kets then in U*psi instead of transposing, conjugating etc...

Reply:
agreed, opened #29